mlir/include/mlir/Dialect/Transform/IR/TransformInterfaces.td

```diff
@@ -280,6 +280,28 @@ def PatternDescriptorOpInterface : OpInterface<"PatternDescriptorOpInterface"> {
   ];
 }
 
+def TypeConversionOpInterface : OpInterface<"TypeConversionOpInterface"> {
+  let description = [{
+    This interface should be implemented by ops that populate type casting
+    of a `transform.cast_and_inline` op. It provides a method to populate a
+    type converter with source/target materialization patterns.
+  }];
+
+  let cppNamespace = "::mlir::transform";
+
+  let methods = [
+    InterfaceMethod<
+      /*desc=*/[{
+        Populate the given type converter with source/target materialization
+        functions.
+      }],
+      /*returnType=*/"void",
+      /*name=*/"populateTypeMaterializations",
+      /*arguments=*/(ins "::mlir::TypeConverter &":$converter)
+    >
+  ];
```

Comment:
Can this somehow be merged with `TypeConverterBuilderOpInterface`? E.g., have an op that creates the converter and sets up materializations correctly. Or modify the existing interface so that it can update a preexisting converter somehow (do we need a converter handle)?

Reply:
We need a way to update a preexisting converter. If I have a conversion for tensors (like in this PR) next to a conversion for `my_dialect_a`, the conversion implementation for `my_dialect_a` shouldn't have to reimplement the one for tensors (essentially because this isn't actually doing dialect conversion, just piggybacking off of the infrastructure for adding conversions based on a type).

We could add `populateTypeMaterializations` to `TypeConverterBuilderOpInterface` with a default implementation that does nothing. Adding a converter handle would only be for if we wanted to reuse the converter with other transforms (which maybe we do?)

Reply:
I combined it with `TypeConverterBuilderOpInterface` and gave the type converter builder a default implementation.